### PR TITLE
Avoids CLI key duplication in executor

### PR DESCRIPTION
**Pull request by mariuszbarczak**
Signed-off-by: Mariusz Barczak <mariusz.barczak@intel.com>

### CODE REVIEW

File: source/octf/cli/Executor.cpp [hunk 1]
```diff
@@ -492,5 +475,32 @@ void Executor::setupOutputsForCommandsLogs() const {
...
+                        longKey);
+    }
+
+    if (!shortKey.empty() && m_localCmdSet->hasCmd(shortKey)) {
+        throw Exception("Cannot add command because of short key conflict: " +
+                        shortKey);
+    }
+
+    for (const auto iter : m_modules) {
+        if (iter.first.getLongKey() == longKey) {
```
Comment (tomaszrybicki) [inline source/octf/cli/Executor.cpp]: as above

File: source/octf/cli/Executor.h [hunk 2]
```diff
@@ -187,6 +187,17 @@ class Executor {
      */
     void setupOutputsForCommandsLogs() const;
 
+    /**
+     * @brief Check if long key or short key already available in executor
```
Comment (tomaszrybicki) [inline source/octf/cli/Executor.h]: is already*

File: source/octf/cli/Executor.h [hunk 3]
```diff
@@ -187,6 +187,17 @@ class Executor {
...
+     *
+     * @param longKey CLI long key to be checked if repeated
+     * @param shortKey CLI long key to be checked if repeated
```
Comment (tomaszrybicki) [inline source/octf/cli/Executor.h]: short - copy paste error

File: source/octf/cli/Executor.cpp [hunk 4]
```diff
@@ -492,5 +475,32 @@ void Executor::setupOutputsForCommandsLogs() const {
     log::cout << log::enable << log::json << log::prefix << prefix;
 }
 
+void Executor::checkIsKeyRepeated(const std::string &shortKey,
+                                  const std::string &longKey) const {
+    if (m_localCmdSet->hasCmd(longKey)) {
```
[the remark this answers is not in the document]
Comment (mariuszbarczak) [reply source/octf/cli/Executor.cpp]: In auto-CLI empty long key is not allowed

File: source/octf/cli/Executor.cpp (hunk 1 above)
Comment (mariuszbarczak) [reply source/octf/cli/Executor.cpp]: In auto-CLI empty long key is not allowed

**Pull request merged**